proto/error/error.proto

```diff
@@ -0,0 +1,30 @@
+syntax = "proto3";
+package nitric.error.v1;
+
+// protoc plugin options for code generation
+option go_package = "nitric/v1;v1";
+option java_package = "io.nitric.proto.error.v1";
+option java_multiple_files = true;
+option java_outer_classname = "Errors";
+option php_namespace = "Nitric\\Proto\\Error\\V1";
+option csharp_namespace = "Nitric.Proto.Error.v1";
+
+// Provides a Service Error Details
+message ErrorDetails {
+
+    // A developer-facing human-readable error message in English. It should
+    // both explain the error and offer an actionable resolution to it.
+    string message = 1;
+
+    // The cause of the error.
+    string cause = 2;
+
+    // The API service invoked, e.g. 'Service.Method'.
+    string service = 3;
+
+    // The plugin method invoked, e.g. 'PluginService.Method'.
+    string plugin = 4;
+
+    // The plugin method arguments, ensure only non-senstivie data is specified.
+    string args = 5;
```

Comment:
This might be better expressed as a kind of scope, that we can expand on without having to introduce breaking changes.

```protobuf
message ErrorScope {
  // XXX: Think about this, is this the level of detail we want?
  // would we want to go deeper?
  string name = 1;
  string plugin = 2;
    
  repeated string args = 3;
}

message ErrorDetails {
  string message = 1;
  string cause = 2;
  ErrorScope scope = 3;
}


```

Reply:
Yep, could do.  A map is a good structure for the plugin arguments, but is more work building up this context information in the error reporting code.  

I think I prefer service to name in the scope structure.

**Existing**

```
[
  code: INTERNAL
  message: DB Fetch error
  cause: not found
  service: DocumentService.Get
  plugin: BoltDocService.Get
  args: [key={Collection: {Name: customer, Parent: <nil>}, Id: medgar.dot.gmail.com}]
]
```

**Revised**

```
[
  code: INTERNAL
  message: DB Fetch error
  cause: not found
  scope: [
    name: DocumentService.Get
    plugin: BoltDocService.Get
    args: [
      key: {Collection: {Name: customer, Parent: <nil>}
      Id: medgar.dot.gmail.com}
    ]
  ] 
]
```